depthai_ros_driver/src/depthai_loader.cpp

```diff
@@ -7,108 +7,51 @@
     }
 
-    dai::Pipeline p;  // = loader->getPipeline();
-    {
-        using namespace dai::node;
-        auto cam = p.create<ColorCamera>();
-        auto out = p.create<XLinkOut>();
-        auto nn1 = p.create<dai::node::NeuralNetwork>();
-        auto nno = p.create<dai::node::XLinkOut>();
-        auto inp = p.create<XLinkIn>();
-        auto ctl = p.create<XLinkIn>();
-
-        out->setStreamName("preview");
-        nno->setStreamName("detections");
-        inp->setStreamName("input_config");
-        ctl->setStreamName("control");
-
-        // Color camera
-        cam->setPreviewSize(300, 300);
-        cam->setResolution(dai::ColorCameraProperties::SensorResolution::THE_1080_P);
-        cam->setInterleaved(false);
-        cam->setColorOrder(dai::ColorCameraProperties::ColorOrder::BGR);
-
-        // object detection
-        nn1->setBlobPath(blob_file);
-
-        inp->out.link(cam->inputConfig);
-        ctl->out.link(cam->inputControl);
-        cam->preview.link(out->input);
-        cam->preview.link(nn1->input);
-        nn1->out.link(nno->input);
-
-        // video output
-        auto xout_video = p.create<XLinkOut>();
-        xout_video->setStreamName("video");
-        cam->video.link(xout_video->input);
-
-        // mono left
-        auto mono_left = p.create<MonoCamera>();
-        auto xout_left = p.create<XLinkOut>();
-        xout_left->setStreamName("left");
-        mono_left->setResolution(dai::MonoCameraProperties::SensorResolution::THE_720_P);
-        mono_left->setBoardSocket(dai::CameraBoardSocket::LEFT);
-        mono_left->out.link(xout_left->input);
-
-        // mono right
-        auto mono_right = p.create<MonoCamera>();
-        auto xout_right = p.create<XLinkOut>();
-        xout_right->setStreamName("right");
-        mono_right->setResolution(dai::MonoCameraProperties::SensorResolution::THE_720_P);
-        mono_right->setBoardSocket(dai::CameraBoardSocket::RIGHT);
-        mono_right->out.link(xout_right->input);
-
-        // stereo
-        auto stereo = p.create<dai::node::StereoDepth>();
-        stereo->setConfidenceThreshold(200);
-        stereo->setRectifyEdgeFillColor(0);
-        stereo->setLeftRightCheck(false);
-        stereo->setExtendedDisparity(false);
-        stereo->setSubpixel(false);
-        mono_left->out.link(stereo->left);
-        mono_right->out.link(stereo->right);
-        stereo->syncedLeft.link(xout_left->input);
-        stereo->syncedRight.link(xout_right->input);
-
-        // depth
-        auto xout_depth = p.create<XLinkOut>();
-        xout_depth->setStreamName("depth");
-        stereo->depth.link(xout_depth->input);
-
-        // disparity
-        auto xout_disparity = p.create<XLinkOut>();
-        xout_disparity->setStreamName("disparity");
-        stereo->disparity.link(xout_disparity->input);
-    }
-
+    auto openvino_version = dai::OpenVINO::Version::VERSION_2020_3;
     rr::DeviceROS driver(openvino_version);
```

Comment:
Will this work with the preview_pipeline and stereo_pipeline?

Reply:
```
    auto openvino_version = dai::OpenVINO::Version::VERSION_2020_3;
    rr::DeviceROS driver(openvino_version);
```
needs to be changed to 
```
    // auto openvino_version = dai::OpenVINO::Version::VERSION_2020_3;
    rr::DeviceROS driver;
```
for `PreviewPipeline` and `StereoPipeline`. The issue is that `openvino` version needs to be set for each `blob` file depending on with which OpenVINO version the blob was generated. 

Any good solution?

Reply:
https://github.com/luxonis/depthai-core/blob/744b32678a0c418dd78cd47736e1eb50a86f9fab/src/device/DeviceBase.cpp#L278-L287

Can we call the init open vino directly, only if the pipeline contains Neural Network?

Reply:
> For pipeline with NN can we call the init open vino directly?
> https://github.com/luxonis/depthai-core/blob/744b32678a0c418dd78cd47736e1eb50a86f9fab/src/device/DeviceBase.cpp#L278-L287
> 

Thanks. I will look into it.

Reply:
openvino version could be controlled by using getter and setter functions of `pipeline`.